### PR TITLE
bpo-37822: Add math.as_integer_ratio().

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue37822

### CODE REVIEW

**Comment by jdemeyer** [comment]
I think it would be good to define more precisely what are the output types. We require the output of `as_integer_ratio` to be two integers, but what does "integer" mean in this context? A Python `int`? An object which implements `__index__`? An instance of `numbers.Integral`?

My personal vote would be to require `__index__` (I don't mean that we should check that condition, but we should document it).

File: Modules/mathmodule.c [hunk 1]
```diff
@@ -3306,9 +3306,83 @@ math_comb_impl(PyObject *module, PyObject *n, PyObject *k)
 }
 
 
+/*[clinic input]
+math.as_integer_ratio
+    x: object
+    /
+greatest common divisor of x and y
```
Comment (brandtbucher) [inline Modules/mathmodule.c:3313]: This looks like a copy-paste error. Maybe:
```suggestion
Return the rational representation of x as the pair (numerator, denominator).
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
One small issue I noticed:

File: Modules/mathmodule.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/mathmodule.c:3313]: Thank you. But this proposition was rejected.

**Pull request closed without merging**